contracts/true-currencies/TrueCurrencyWithPoR.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.10;
+
+import {TrueCurrency} from "./TrueCurrency.sol";
+import {IChainlinkAggregatorV3} from "../common/interface/IChainlinkAggregatorV3.sol";
+import {IPoRToken} from "../common/interface/IPoRToken.sol";
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+
+/**
+ * @title TrueCurrencyWithPoR
+ * @dev TrueCurrencyPoR is an ERC20 with blacklist & redemption addresses.
+ *  Please see TrueCurrency for the implementation that this contract inherits from.
+ *  This contract implements an additional check against a Proof-of-Reserves feed before
+ *  allowing tokens to be minted.
+ */
+abstract contract TrueCurrencyWithPoR is TrueCurrency, IPoRToken {
+    using SafeMath for uint256;
+
+    uint256 public constant MAX_AGE = 7 days;
+
+    constructor() public {
+        chainReserveHeartbeat = MAX_AGE;
+    }
+
+    /**
+     * @notice Overriden mint function that checks the specified proof-of-reserves feed to
+     * ensure that the total supply of this TrueCurrency is not greater than the reported
+     * reserves.
+     * @dev The proof-of-reserves check is bypassed if feed is not set.
+     * @param account The address to mint tokens to
+     * @param amount The amount of tokens to mint
+     */
+    function _mint(address account, uint256 amount) internal virtual override {
+        if (chainReserveFeed == address(0)) {
+            super._mint(account, amount);
+            return;
+        }
+
+        // Get latest proof-of-reserves from the feed
+        (, int256 signedReserves, , uint256 updatedAt, ) = IChainlinkAggregatorV3(chainReserveFeed).latestRoundData();
+        require(signedReserves > 0, "TrueCurrency: Invalid answer from PoR feed");
```

Comment:
Just a small suggestion to save gas but maybe we can shorten the `require` messages